src/recommendpolicy/helperFunctions.go

```diff
@@ -118,3 +124,46 @@ func addPolicyRule(policy *types.KnoxSystemPolicy, r *types.KnoxSystemSpec) {
 	}
 
 }
+
+func initDeploymentWatcher() {
+	clientset := cluster.ConnectK8sClient()
+	watcher, err := clientset.AppsV1().Deployments("").Watch(context.TODO(), metav1.ListOptions{})
+
+	if err != nil {
+		log.Error().Msg(err.Error())
+		return
+	}
+	defer watcher.Stop()
+
+	for event := range watcher.ResultChan() {
+		found := false
+		var index int
+		switch event.Type {
+		case watch.Added:
+			for _, data := range DeployNsName {
+				if data.Name == event.Object.(*v1.Deployment).Name && data.Namespace == event.Object.(*v1.Deployment).Namespace {
+					found = true
+					break
+				}
+			}
+			if !found {
+				log.Info().Msgf("Found Deployment %v in %v namespace", event.Object.(*v1.Deployment).Name, event.Object.(*v1.Deployment).Namespace)
+				generateHardenPolicy(event.Object.(*v1.Deployment).Name, event.Object.(*v1.Deployment).Namespace, event.Object.(*v1.Deployment).Spec.Template.Labels)
```

Comment:
Are we extracting recommended policies based only on deployment name & labels
What if the same contains multiple containers ?? We are not getting recommended policies for containers ?